### ISSUE TITLE
Please allow disabling automatic .gitignore generation; or any arbitrary .tpl file

### ISSUE THREAD

**Issue by mkeveney** [platformio/platformio-core#4181]
My platformIO projects generally live in a subfolder of a larger .git repository.
I prefer to manage a single .gitignore file in the root of my repository.

PlatformIO insists upon re-generatng the .gitignore file.  I'd like to be able to turn this off.

This subject is discussed here: 
    https://community.platformio.org/t/how-to-disable-auto-generating-gitignore-when-using-vscode/17125

Where @maxgearhardt helpfully detailed the technical reasons for this.  He suggests a more general solution might be to allow disabling _any_ .tpl file.

-Matt

**Comment by ivankravets** [comment]
Thanks for the issue. We will think about how to fix it.

A temporary solution is to use [Advanced Scripting](https://docs.platformio.org/en/latest/scripting/index.html) and extra script:
```python
import os

if os.path.isfile(".gitignore"):
    os.remove(".gitignore")
```

**Comment by CircadianRebel** [comment]
Came here to file this same issue. My organization uses hg, not git, and this bad behavior on pio's part is a pain and causes usability issues. e.g. it causes programs such as vs code to constantly think there are modified files.

**Comment by dazca** [comment]
The automatic .gitignore creation should be disabled by default. Creating a .gitignore is bound to those who use git, and is not mandatory to use git to use PlatformIO.

**Comment by NathanSweet** [comment]
I'm using the script workaround but `extra_scripts` goes under `env`, so the `.gitignore` file will still be created and hang out until the next build is run.

**Comment by McKayRansom** [comment]
I also ran into this issue, generating a .gitignore is not intuitive behavior. Took me a while to figure out it was PlatformIO. Would love an option in the VS Code extension to turn this off. That way it would show up when searching for VSCode settings. That was the first place I looked.

**Comment by NilsNachname** [comment]
Yes please. This conflicts with my project structure and it's very bothersome that there is no way to disable the behavior.

**Comment by arduhe** [comment]
I have a - admittedly somewhat rough 😉  - solution for this problem. I have deleted the annoying template, which could be found under 

`~/.platformio/penv/lib/python3.6/site-packages/platformio/project/integration/tpls/vscode/.gitignore.tpl`

Nevertheless, I would prefer a more permanent and less invasive solution

**Comment by ElTrasho3000** [comment]
I also would highly appreciate the automatic generation of .gitignore to not happen as default. Meanwhile I'll use the workaround.

**Comment by mkeveney** [comment]
@arduhe 
> I have a - admittedly somewhat rough 😉 - solution for this problem. I have deleted the annoying template
Actually, I like this much better than the script.  Thanks for the idea!

**Comment by IlyaMatsuev** [comment]
Another "solution" which was surprisingly not mentioned anywhere - just ignore ".gitgnore" in all subdirectories from your main ".gitignore" file.

It will still be generated but it wouldn't be annoyingly appearing as an untracked file in your changes.
